src/pages/api/paymentTxParams/index.ts

```diff
@@ -1,100 +1,31 @@
+import { createPaymentTxOrMsg } from '@/services/paymentTxOrMsgService';
+import { Payload } from '@/types/paymentTx';
 import { NextApiRequest, NextApiResponse } from 'next';
-import { getPrismaClient } from '@helpers/database';
-import { v4 as uuidv4 } from 'uuid';
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  const prisma = getPrismaClient();
-
   if (req.method === 'POST') {
     try {
-      const { 
-        payloadType,
-        uuid,
-        chainId, 
-        dappUrl, 
-        dappName,
-        rpcProxySubmissionParams, 
-
-        // fields relevant to the eip712 payload type
-        message,
-        additionalPayload,
-        
-        // fields relevant to contractCall payload type
-        requiresSenderAddress, 
-        contractAbi, 
-        placeholderSenderAddress, 
-        approveTxs, 
-        paymentTx, 
-
-        // fields relevant to eip681 payload type
-        contractAddress,
-        toAddress,
-        value,
-      } = req.body;
-
-      if (!payloadType) {
-        return res.status(400).json({ message: 'Payload type is required' });
+      // validations
+      const payloadType = req.body.payloadType;
+      if (!payloadType || !['eip681', 'contractCall', 'eip712'].includes(payloadType)) {
+        return res.status(400).json({ message: 'Invalid or missing payload type' });
       }
 
-      if (!['eip681', 'contractCall', 'eip712'].includes(payloadType)) {
-        return res.status(400).json({ message: 'Invalid payload type, must be eip681 or contractCall' });
-      }
-
-      const paymentUuid = uuid || uuidv4();
```

Comment:
Moved all these checks to the service, and defined proper types for them. IN case we ever need to deserialize (asked Cursor for help)